### PR TITLE
Fix/dialog swiping does not trigger on dialog dismissed

### PR DESCRIPTION
**Pull request by M-i-k-e-l**
## Description
Dialog - fix swiping to dismiss does not trigger onDialogDismissed

## Changelog
Dialog - fix swiping to dismiss does not trigger onDialogDismissed

### CODE REVIEW

File: src/components/dialog/index.tsx [hunk 1]
```diff
@@ -182,6 +183,19 @@ class Dialog extends Component<DialogProps, DialogState> {
         _.invoke(props, 'onDialogDismissed', props);
       }
     });
+  }
+
+  onDismiss = () => {
+    let fadeOut = false;
+    if (Constants.isIOS && this.props.visible) {
+      fadeOut = true;
+    }
```
Comment (ethanshar) [inline src/components/dialog/index.tsx:192]: Try that.. 
```suggestion
    const fadeOut = Constants.isIOS && this.props.visible;
```

File: src/components/dialog/index.tsx [hunk 2]
```diff
@@ -171,8 +172,8 @@ class Dialog extends Component<DialogProps, DialogState> {
     }
   }
 
-  onDismiss = () => {
-    this.setState({modalVisibility: false}, () => {
+  _onDismiss = () => {
+    this.setState({modalVisibility: false, fadeOut: false}, () => {
```
Comment (ethanshar) [inline src/components/dialog/index.tsx:176]: I didn't completely understand the `fadeOut` fix.. 
Also, It seem the dialog works as expected when dismissing it using swipe (on master) 
Maybe we should sync about it..

Comment (M-i-k-e-l) [reply src/components/dialog/index.tsx:176]: Did you test it in public or private?

Comment (ethanshar) [reply src/components/dialog/index.tsx:176]: hmm wait.. 
Wasn't the issue that after dismissing using swipe the Dialog doesn't show the next time?

Comment (M-i-k-e-l) [reply src/components/dialog/index.tsx:176]: ActionSheet (private)

Comment (ethanshar) [reply src/components/dialog/index.tsx:176]: So shouldn't the fix be part of ActionSheet code?

Comment (M-i-k-e-l) [reply src/components/dialog/index.tsx:176]: No, it's the same as the previous PR to the Dialog for RN63, it fixes the `onDialogDismissed` API

File: src/components/dialog/index.tsx (hunk 1 above)
Comment (M-i-k-e-l) [reply src/components/dialog/index.tsx:192]: 10x :)

**Review by ethanshar — APPROVED** [review]

**Pull request merged**